### PR TITLE
Elevate `performAndWaitOrThrow` to public API

### PR DESCRIPTION
**Pull request by zwaldowski**

### CODE REVIEW

File: CoreDataStack/NSManagedObjectContext+AsyncHelpers.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import CoreData
+import Swift
+
+extension NSManagedObjectContext {
+    /**
+     Synchronously exexcutes a given function on the receiver’s queue.
+
+     You use this method to safely address managed objects on a concurrent
+     queue. This method may safely be called reentrantly.
```
Comment (rcedwards) [inline CoreDataStack/NSManagedObjectContext+AsyncHelpers.swift]: looks good.

add `throws`, 'parameter`, and 'returns' to the doc header like these https://github.com/bignerdranch/CoreDataStack/blob/master/CoreDataStack/CoreDataModelable.swift#L53

File: CoreDataStack/NSPersistentStoreCoordinator+AsyncHelpers.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import CoreData
+
+extension NSPersistentStoreCoordinator {
+    /**
+     Synchronously exexcutes a given function on the coordinator's internal
+     queue.
+
+     This method may safely be called reentrantly.
+    **/
```
Comment (rcedwards) [inline CoreDataStack/NSPersistentStoreCoordinator+AsyncHelpers.swift]: Same thing here for method docs

**Comment by zwaldowski** [comment]
Doc commends addressed

**Comment by rcedwards** [comment]
:+1:

**Pull request merged**